File: src/main/java/com/example/timetable/profile/ProfileEntity.java
package com.example.timetable.profile;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
@Getter
@Setter
@Entity
public class ProfileEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(name = "name")
    private String name;
    @Column(name = "surname")
    private String surname;
    @Column(name = "email")
    private String email;
    @Column(name = "password")
    private String password;
    @Enumerated(EnumType.STRING)
    @Column(name = "role")
    private ProfileRole role;
}

File: src/main/kotlin/com/example/w3schooluz/profile/ProfileEntity.kt
package com.example.w3schooluz.profile

import jakarta.persistence.*
@Entity
@Table(name = "profile")
class ProfileEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int? = null,

    @Column(name = "name")
    var name: String? = null,

    @Column(name = "surname")
    var surname: String? = null,

    @Column(name = "email")
    val email: String? = null,

    @Column(name = "phone")
    var phone: String? = null,

    @Column(name = "password")
    var password: String? = null,

    @Enumerated(EnumType.STRING)
    @Column(name = "role")
    var role: ProfileRole? = null,

    @Enumerated(EnumType.STRING)
    @Column(name = "status")
    val status: ProfileStatus? = null
)